falcon/util/misc.py

```diff
@@ -305,3 +311,89 @@ def get_http_status(status_code, default_reason='Unknown'):
     except AttributeError:
         # not found
         return str(code) + ' ' + default_reason
+
+
+@functools.lru_cache(maxsize=64)
```

Comment:
I've commented on https://github.com/falconry/falcon/issues/1135#issuecomment-550187832, `functools.lru_cache` had thread-safety issues which where fixed post-3.6 release.
Maybe the issue is less serious here with async being mixed-and-matched with threads less often, and us mandating Python 3.6+, but still thought it might be worth bringing attention to the issue. Maybe we should check which CPython 3.5.x and 3.6.x versions are susceptible to the issue, and make a decision based on the findings.

Reply:
Good catch. I think we will have to do something about this.

Reply:
I opted to just use a noop for older pythons. The performance decrease probably won't be noticeable for most people, and over time those platforms will be used less and less.